File: Programmers/Python/신규 아이디 추천.py
def solution(new_id):
    new_id = new_id.lower()
    
    p = re.compile('[^a-z0-9-_.]')
    new_id = p.sub('',new_id)
    p = re.compile('[.]{2,}')
    new_id = p.sub('.',new_id)
    p = re.compile('^[.]|[.]$')
    new_id = p.sub('',new_id)
    
    new_id = 'a' if len(new_id) == 0 else new_id[0:15]
    new_id = p.sub('',new_id)
    if len(new_id) <= 2:
        c = new_id[len(new_id)-1]
        while len(new_id) < 3:
            new_id += c
            
    return new_id

File: Programmers/Java/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        new_id = new_id.toLowerCase();
        new_id = new_id.replaceAll("[^a-z0-9-_.]","");	
    	  new_id = new_id.replaceAll("[.]{2,}",".");
    	  new_id = new_id.replaceAll("^[.]|[.]$",""); 
    	
        if(new_id.equals(""))
            new_id = "a";
        
        int len = new_id.length();
        if(len >= 16){
            new_id = new_id.substring(0,15);
            new_id = new_id.replaceAll("[.]$","");
        }
    
        len = new_id.length();
        if(len <= 2){
            char c = new_id.charAt(len-1);
            while(new_id.length() < 3)
                new_id += c;
        }
        
        return new_id;
    }
}
